File: client/rejsy-koalicja/src/shared/models/charterDto.ts
import {Equipment} from "../enums/equipment";
import {CaptainDto} from "./captainDto";
import {PayoffDto} from "./payoffDto";

export default interface CharterDto {
    equipment?: Equipment[]
    from: Date
    to: Date
    captain?: CaptainDto
    crewNumber?: number
    payoff?: PayoffDto
    reservation?: Date
    thermsAndConditionVersion?: string
}

File: server/src/main/java/pl/rejsykoalicja/charter/dto/CharterDto.java
package pl.rejsykoalicja.charter.dto;

import lombok.Builder;
import lombok.Data;
import pl.rejsykoalicja.charter.enums.Equipment;

import java.time.ZonedDateTime;
import java.util.List;

@Builder
@Data
public class CharterDto {
    private List<Equipment> equipment;
    private ZonedDateTime from;
    private ZonedDateTime to;
    private CaptainDto captain;
    private Integer crewNumber;
    private PayoffDto payoff;
    private ZonedDateTime reservation;
    private String thermsAndConditionVersion;
}
